community/codegen/src/main/java/org/neo4j/codegen/source/ClasspathHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.codegen.source;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public final class ClasspathHelper
+{
+    private ClasspathHelper()
+    {
+        throw new AssertionError( "Not for instantiation!" );
+    }
+
+    public static String javaClasspathString()
+    {
+        return System.getProperty( "java.class.path" );
+    }
+
+    public static Set<String> javaClasspath()
+    {
+        String[] classpathElements = javaClasspathString().split( ":" );
+        Set<String> result = new LinkedHashSet<>( classpathElements.length, 1 );
```

Comment:
`1` is a very high load factor, the hash table of the map will suffer a very high degree of collisions.